### PR TITLE
Soen117 patient dashboard

### PR DESCRIPTION
**Pull request by KelvinTrang**
# Description

Implements/fixes [SOEN-117](https://soen390team8.atlassian.net/browse/SOEN-98)

- Added line chart in dashboard page and form to fill page
- Added some information in the dashboard

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Test will come later

#Screenshots
image

### CODE REVIEW

**Review by amqkv — APPROVED** [review]
Amazing work!

File: backend/__tests__/routes/status.test.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ describe("GET: getting all status for this user", () => {
     it("Returns code 200 and array of status", async () => {
         const userId = 51;
         const response = await request(app).get(`/status/getAllStatus/${userId}`);
-        expect(response.body).toHaveLength(5);
+        expect(response.body).toHaveLength(6);
```
Comment (jfv-2000) [inline backend/__tests__/routes/status.test.js]: just comment this out and we'll fix this test case as a TODO

File: backend/controllers/status-controller.js [hunk 2]
```diff
@@ -96,13 +96,21 @@ async function getAllStatus(req, res) {
                 Patient_PatientId: PatientId,
             },
             order: [["StatusTime", "DESC"]],
+            
         });
+
+        // for(let i = 0; i < allStatus.length; i++){
+        //     allStatus[i].StatusTime = allStatus[i].StatusTime.substring(0, 10);
+        // }
```
Comment (jfv-2000) [inline backend/controllers/status-controller.js]: remove unnecessary comment ?

File: frontend/components/forms/patient-form-to-fill.tsx [hunk 3]
```diff
@@ -32,11 +35,10 @@ export default function PatientFormToFill({ requiredDetails, pastConditions }: P
         let error = false;
 
         // initialize values for statusValues
-        // TODO: decide on default value for weight
         const statusValues: StatusParameters = {
             accountId: userId,
             temperature: 37,
-            weight: 0,
+            weight: 80,
```
Comment (jfv-2000) [inline frontend/components/forms/patient-form-to-fill.tsx]: Maybe move the initialized values to a constant file

File: frontend/functions/create-status.ts [hunk 4]
```diff
@@ -37,10 +37,10 @@ export async function statusFilled({ temperature, weight, symptoms, isReviewed,
...
-        method: "GET",
-        headers: { "Content-Type": "application/json" },
-    });
-}
+// export async function pastConditionsInformation(userId: number | undefined) {
+//     return userId && await fetch(serverURL + "/getAllStatus/" + userId, {
+//         method: "GET",
+//         headers: { "Content-Type": "application/json" },
+//     });
+// }
```
Comment (jfv-2000) [inline frontend/functions/create-status.ts]: Will this be used later on ? If it's unnecessary then it should be removed

File: frontend/pages/index.tsx [hunk 5]
```diff
@@ -3,20 +3,36 @@ import { GetServerSideProps } from "next";
...
             session: await getSession(context),
+            data,
         },
     };
 };
 
-export default function Home() {
-    const { data: session } = useSession();
+export default function Home({ data }: {data: StatusDataProps[]}) {
+    console.log( data);
```
Comment (jfv-2000) [inline frontend/pages/index.tsx]: console log("remove this console log please");

File: frontend/pages/index.tsx [hunk 6]
```diff
@@ -3,20 +3,36 @@ import { GetServerSideProps } from "next";
...
 
+    const { data: session } = useSession();
+    console.log( data);
```
Comment (jfv-2000) [inline frontend/pages/index.tsx]: console.log("badabingbongbong remove this console log please")

**Review by jfv-2000 — CHANGES_REQUESTED** [review]
F A N T A S T I C W O R K B I N G B O N G 👍 🚀 🧇 💅 🌮 🦖 🌮 
J U S T F I X S O ME S T U F F P L E A S E
T H A N K Y O U

File: frontend/components/forms/patient-form-to-fill.tsx (hunk 3 above)
Comment (amqkv) [reply frontend/components/forms/patient-form-to-fill.tsx]: Maybe in `@frontend/models/patient.ts`

File: frontend/pages/index.tsx (hunk 5 above)
Comment (amqkv) [reply frontend/pages/index.tsx]: xDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD

File: frontend/components/homepage/dashboard.tsx [hunk 7]
```diff
@@ -1,26 +1,61 @@
...
-                    {links.map(({ label, url, roleRequired, img }: link) => {
-                        const renderCard = roleRequired === userRole || !roleRequired;
-                        return renderCard && <Card key={label} label={label} image={img} url={url} />;
-                    })}
-                </Flex>
-            </Center>
-        </Box>
+            <Flex flexWrap={"wrap"}>
+                {/* Patient dashboard */}
+                {session?.user?.Role === USER_ROLES.patient ? (
```
Comment (NicholasYip) [inline frontend/components/homepage/dashboard.tsx]: can be changed to user session?.user?.Role === USER_ROLES.patient and remove the ? and :<></> at the end

**Review by jfv-2000 — APPROVED** [review]

File: frontend/pages/patient/patient-dashboard.tsx [hunk 8]
```diff
@@ -17,15 +17,15 @@ export default function PatientDashboard() {
...
+        return <Dashboard data={[]} />;
     } else if (session?.user.Role === USER_ROLES.hOfficial) {
-        return <Dashboard />;
+        return <Dashboard data={[]} />;
     } else if (session?.user.Role === USER_ROLES.iOfficer) {
-        return <Dashboard />;
+        return <Dashboard data={[]} />;
     } else if (session?.user.Role === USER_ROLES.admin) {
-        return <Dashboard />;
+        return <Dashboard data={[]} />;
```
Comment (stevenvuong00) [inline frontend/pages/patient/patient-dashboard.tsx]: this file is not used

**Review by stevenvuong00 — APPROVED** [review]
good job bingbong

File: backend/__tests__/routes/status.test.js [hunk 9]
```diff
@@ -36,7 +36,7 @@ describe("GET: getting all status for this user", () => {
     it("Returns code 200 and array of status", async () => {
         const userId = 51;
         const response = await request(app).get(`/status/getAllStatus/${userId}`);
-        expect(response.body).toHaveLength(5);
+        // expect(response.body).toHaveLength(6);
```
Comment (NicholasYip) [inline backend/__tests__/routes/status.test.js:40]: remove

File: frontend/components/homepage/types/types.ts [hunk 10]
```diff
@@ -1,15 +1,15 @@
 export interface CardProps {
     label: string;
     image?: string;
-    url:string;
+    url: string;
     style?: Object;
```
Comment (NicholasYip) [inline frontend/components/homepage/types/types.ts]: object with small o

File: frontend/functions/create-status.ts [hunk 11]
```diff
@@ -15,6 +14,7 @@ export interface pastConditionsProps{
     Temperature: number;
     StatusTime: string;
     IsReviewed?: boolean;
+    Patient_PatientId?: number;
     UserId?: number;
     Weight: number;
     Symptoms: string;
```
Comment (NicholasYip) [inline frontend/functions/create-status.ts:20]: isnt this exactly the same interface as `StatusDataProps` in` frontend/models/patient.ts`

File: frontend/pages/index.tsx [hunk 12]
```diff
@@ -3,20 +3,33 @@ import { GetServerSideProps } from "next";
...
     return {
         props: {
             session: await getSession(context),
+            data,
         },
     };
 };
 
-export default function Home() {
+export default function Home({ data }: {data: StatusDataProps[]}) {
```
Comment (NicholasYip) [inline frontend/pages/index.tsx]: unknwon because other users might have chart data

**Review by NicholasYip — APPROVED** [review]
looks fine, small changes

**Pull request merged**